src/pages/index.js

```diff
@@ -1,168 +1,241 @@
+  selectors.profilePopupSelector,
+  handleProfileEditSubmit
+);
+const addCardPopup = new PopupWithForm(
+  selectors.cardPopupSelector,
+  handleCardAddClick
+);
+const cardSelector = selectors.cardTemplate;
 const userInfo = new UserInfo({
   userNameSelector: ".profile__title",
-
   userDescriptionSelector: ".profile__description",
+  userAvi: ".profile__image",
 });
 
 /* -------------------------------------------------------------------------- */
-
-/*                              POPUP WITH IMAGE                              */
-
+/*                                  FUNCTIONS & HANDLERS                      */
 /* -------------------------------------------------------------------------- */
-const previewImagePopup = new PopupWithImage("#modal-preview");
-previewImagePopup.setEventListeners();
+let newCardSection;
+let userId;
 
-/* -------------------------------------------------------------------------- */
-
-/*                               POPUP WITH FORM                              */
-
-/* -------------------------------------------------------------------------- */
-
-/* --------------------------- edit profile modal --------------------------- */
-
-const profileEditButton = document.querySelector(".profile__edit-button");
-
-const profileEditPopup = new PopupWithForm(
-  "#profile-edit-modal",
-
-  handleProfileEditSubmit
-);
-
-profileEditPopup.setEventListeners();
+function createCard(cardData) {
+  const card = new Card(
+    cardData.name,
+    cardData.link,
+    cardData.isLiked,
+    cardData.likes,
+    cardData._id,
+    userId,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    cardSelector
+  );
 
-profileEditButton.addEventListener("click", () => {
-  handleProfileEditClick();
-});
+  return card.getView();
+}
+function handleProfileEditSubmit(inputValues) {
+  profileEditPopup.renderLoading(true);
+  api
+    .updateProfileInfo(inputValues)
+    .then(() => {
+      userInfo.setUserInfo(inputValues);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false, "Save");
+    });
+  console.log(inputValues);
+}
+function handleCardAddClick(inputValues) {
+  addCardPopup.renderLoading(true);
+  api
+    .addNewCard(inputValues)
+    .then((cardData) => {
+      const addCard = createCard(cardData);
+      newCardSection.addItem(addCard);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false, "Create");
+    });
+}
+function handleAviPopup(inputValues) {
+  avatarPopup.renderLoading(true);
+  api
+    .updateProfileAvatar(inputValues.avatar)
+    .then(() => {
+      userInfo.setAvatar(inputValues.avatar);
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarPopup.renderLoading(false);
```

Comment:
* When I edit the avatar the button text disappears https://disk.yandex.ru/i/Fqhwun2BOPDfSA 


```js
      avatarPopup.renderLoading(false, "Save");


```